app/mailers/patient_mailer.rb

```diff
@@ -55,6 +55,9 @@ def assessment_sms_weblink(patient)
       return
     end
 
+    # Cover potential race condition where multiple messages are sent for the same monitoree.
+    return unless patient.last_assessment_reminder_sent_eligible?
```

Comment:
Agreed, it's worth revisiting this as a holistic view of how the patient mailer works. At the moment I think it's fine, since it's pretty clear what these guards are doing.